### ISSUE TITLE
[libgnutls] Build error when cross-compiling for mingw-w64

### ISSUE THREAD

**Issue by comex** [microsoft/vcpkg#32085]
Package: libgnutls:x64-mingw-static -> 3.7.8

**Host Environment**

- Host: arm64-osx
- Compiler: GNU 13.1.0
-    vcpkg-tool version: 2023-06-15-8c254a5fb6c503615834fc66bd0717664a339a2e
    vcpkg-scripts version: c7519de0c 2023-06-17 (16 hours ago)
- MinGW-w64 environment installed via MacPorts:
```
  x86_64-w64-mingw32-binutils @2.40_1 (active)
  x86_64-w64-mingw32-crt @11.0.1_0 (active)
  x86_64-w64-mingw32-gcc @13.1.0_0 (active)
  x86_64-w64-mingw32-headers @11.0.1_0 (active)
  x86_64-w64-mingw32-widl @11.0.1_0 (active)
  x86_64-w64-mingw32-winpthreads @11.0.1_0 (active)
```
- Same issue when building on an x86_64 Linux machine with MinGW installed from Debian (GCC 12.2.0-14+25.2, MinGW-w64 10.0.0-3)

**To Reproduce**

Starting with nothing installed:
`vcpkg install libgnutls:x64-mingw-static`

**Failure logs**

```
-- Downloading https://gnupg.org/ftp/gcrypt/gnutls/v3.7/gnutls-3.7.8.tar.xz;https://mirrors.dotsrc.org/gcrypt/gnutls/v3.7/gnutls-3.7.8.tar.xz;https://www.mirrorservice.org/sites/ftp.gnupg.org/gcrypt/gnutls/v3.7/gnutls-3.7.8.tar.xz -> gnutls-3.7.8.tar.xz...
-- Extracting source /private/tmp/vcpkg/downloads/gnutls-3.7.8.tar.xz
-- Applying patch use-gmp-pkgconfig.patch
-- Using source at /private/tmp/vcpkg/buildtrees/libgnutls/src/v3.7.8-02e889560f.clean
-- Found external ninja('1.11.1').
-- Getting CMake variables for x64-mingw-static-dbg
-- Getting CMake variables for x64-mingw-static-rel
-- Generating configure for x64-mingw-static
-- Finished generating configure for x64-mingw-static
-- Configuring x64-mingw-static-dbg
-- Configuring x64-mingw-static-rel
-- Building x64-mingw-static-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/bin/make V=1 -j 11 -f Makefile all
    Working Directory: /private/tmp/vcpkg/buildtrees/libgnutls/x64-mingw-static-dbg/
    See logs for more information:
      /private/tmp/vcpkg/buildtrees/libgnutls/build-x64-mingw-static-dbg-out.log
      /private/tmp/vcpkg/buildtrees/libgnutls/build-x64-mingw-static-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:151 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:2 (vcpkg_build_make)
  ports/libgnutls/portfile.cmake:58 (vcpkg_install_make)
  scripts/ports.cmake:147 (include)



```
[/private/tmp/vcpkg/buildtrees/libgnutls/build-x64-mingw-static-dbg-out.log](https://gist.github.com/comex/fdf771733ca1905a23407b53e429180b)

<details><summary>/private/tmp/vcpkg/buildtrees/libgnutls/build-x64-mingw-static-dbg-err.log</summary>

```
../.././../src/v3.7.8-02e889560f.clean/lib/x509/output.c: In function 'print_time':
../.././../src/v3.7.8-02e889560f.clean/lib/x509/output.c:445:47: warning: format '%lu' expects argument of type 'long unsigned int', but argument 3 has type 'time_t' {aka 'long long int'} [-Wformat=]
  445 |                 addf(str, "error: gmtime_r (%lu)\n", timestamp);
      |                                             ~~^      ~~~~~~~~~
      |                                               |      |
      |                                               |      time_t {aka long long int}
      |                                               long unsigned int
      |                                             %llu
../.././../src/v3.7.8-02e889560f.clean/lib/x509/output.c:450:47: warning: format '%lu' expects argument of type 'long unsigned int', but argument 3 has type 'time_t' {aka 'long long int'} [-Wformat=]
  450 |                 addf(str, "error: strftime (%lu)\n", timestamp);
      |                                             ~~^      ~~~~~~~~~
      |                                               |      |
      |                                               |      time_t {aka long long int}
      |                                               long unsigned int
      |                                             %llu
../.././../src/v3.7.8-02e889560f.clean/lib/x509/verify-high2.c: In function 'load_dir_certs':
../.././../src/v3.7.8-02e889560f.clean/lib/x509/verify-high2.c:442:26: warning: cast increases required alignment of target type [-Wcast-align]
  442 |         dirp = _topendir((_TCHAR*)utf16.data);
      |                          ^
In file included from ../.././../src/v3.7.8-02e889560f.clean/lib/auth/../str.h:29,
                 from ../.././../src/v3.7.8-02e889560f.clean/lib/auth/../gnutls_int.h:305,
                 from ../.././../src/v3.7.8-02e889560f.clean/lib/auth/vko_gost.c:22:
../.././../src/v3.7.8-02e889560f.clean/lib/auth/vko_gost.c: In function 'print_priv_key':
../.././../src/v3.7.8-02e889560f.clean/lib/auth/vko_gost.c:111:26: warning: unknown conversion type character 'z' in format [-Wformat=]
  111 |         _gnutls_hard_log("INT: VKO PRIVATE KEY[%zd]: %s\n",
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../.././../src/v3.7.8-02e889560f.clean/lib/auth/../errors.h:137:23: note: in definition of macro 'LEVEL'
  137 |       _gnutls_log( l, __VA_ARGS__); } while(0)
      |                       ^~~~~~~~~~~
../.././../src/v3.7.8-02e889560f.clean/lib/auth/vko_gost.c:111:9: note: in expansion of macro '_gnutls_hard_log'
  111 |         _gnutls_hard_log("INT: VKO PRIVATE KEY[%zd]: %s\n",
      |         ^~~~~~~~~~~~~~~~
../.././../src/v3.7.8-02e889560f.clean/lib/auth/vko_gost.c:111:49: note: format string is defined here
  111 |         _gnutls_hard_log("INT: VKO PRIVATE KEY[%zd]: %s\n",
      |                                                 ^
../.././../src/v3.7.8-02e889560f.clean/lib/auth/vko_gost.c:111:26: warning: format '%s' expects argument of type 'char *', but argument 3 has type 'size_t' {aka 'long long unsigned int'} [-Wformat=]
  111 |         _gnutls_hard_log("INT: VKO PRIVATE KEY[%zd]: %s\n",
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  112 |                          bytes, _gnutls_bin2hex(priv_buf,
      |                          ~~~~~
      |                          |
      |                          size_t {aka long long unsigned int}
../.././../src/v3.7.8-02e889560f.clean/lib/auth/../errors.h:137:23: note: in definition of macro 'LEVEL'
  137 |       _gnutls_log( l, __VA_ARGS__); } while(0)
      |                       ^~~~~~~~~~~
../.././../src/v3.7.8-02e889560f.clean/lib/auth/vko_gost.c:111:9: note: in expansion of macro '_gnutls_hard_log'
  111 |         _gnutls_hard_log("INT: VKO PRIVATE KEY[%zd]: %s\n",
      |         ^~~~~~~~~~~~~~~~
../.././../src/v3.7.8-02e889560f.clean/lib/auth/vko_gost.c:111:55: note: format string is defined here
  111 |         _gnutls_hard_log("INT: VKO PRIVATE KEY[%zd]: %s\n",
      |                                                      ~^
      |                                                       |
      |                                                       char *
      |                                                      %lld
../.././../src/v3.7.8-02e889560f.clean/lib/auth/vko_gost.c:111:26: warning: too many arguments for format [-Wformat-extra-args]
  111 |         _gnutls_hard_log("INT: VKO PRIVATE KEY[%zd]: %s\n",
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../.././../src/v3.7.8-02e889560f.clean/lib/auth/../errors.h:137:23: note: in definition of macro 'LEVEL'
  137 |       _gnutls_log( l, __VA_ARGS__); } while(0)
      |                       ^~~~~~~~~~~
../.././../src/v3.7.8-02e889560f.clean/lib/auth/vko_gost.c:111:9: note: in expansion of macro '_gnutls_hard_log'
  111 |         _gnutls_hard_log("INT: VKO PRIVATE KEY[%zd]: %s\n",
      |         ^~~~~~~~~~~~~~~~
../.././../src/v3.7.8-02e889560f.clean/lib/auth/srp_sb64.c: In function '_gnutls_sbase64_encode':
../.././../src/v3.7.8-02e889560f.clean/lib/auth/srp_sb64.c:183:17: warning: use of uninitialized value '&tmpres' [CWE-457] [-Wanalyzer-use-of-uninitialized-value]
  183 |                 memcpy(&(*result)[j], tmpres, tmp);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  'gnutls_srp_base64_encode2': events 1-2
    |
    |  349 | gnutls_srp_base64_encode2(const gnutls_datum_t * data,
    |      | ^~~~~~~~~~~~~~~~~~~~~~~~~
    |      | |
    |      | (1) entry to 'gnutls_srp_base64_encode2'
    |......
    |  355 |         size = _gnutls_sbase64_encode(data->data, data->size, &res);
    |      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    |      |                |
    |      |                (2) calling '_gnutls_sbase64_encode' from 'gnutls_srp_base64_encode2'
    |
    +--> '_gnutls_sbase64_encode': events 3-8
           |
           |  141 | _gnutls_sbase64_encode(uint8_t * data, size_t data_size, char **result)
           |      | ^~~~~~~~~~~~~~~~~~~~~~
           |      | |
           |      | (3) entry to '_gnutls_sbase64_encode'
           |......
           |  145 |         uint8_t tmpres[4];
           |      |                 ~~~~~~
           |      |                 |
           |      |                 (4) region created on stack here
           |      |                 (5) capacity: 4 bytes
           |......
           |  157 |         if ((*result) == NULL)
           |      |            ~
           |      |            |
           |      |            (6) following 'false' branch...
           |......
           |  160 |         i = j = 0;
           |      |             ~~~~~
           |      |               |
           |      |               (7) ...to here
           |......
           |  163 |         if (mod > 0) {
           |      |            ~
           |      |            |
           |      |            (8) following 'false' branch (when 'mod == 0')...
           |
         '_gnutls_sbase64_encode': event 9
           |
           |cc1:
           | (9): ...to here
           |
         '_gnutls_sbase64_encode': events 10-14
           |
           |  177 |         for (; i < data_size; i += 3, j += 4) {
           |      |                ~~^~~~~~~~~~~
           |      |                  |
           |      |                  (10) following 'true' branch...
           |  178 |                 tmp = encode(tmpres, &data[i], data_size - i);
           |      |                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           |      |                       |
           |      |                       (11) ...to here
           |  179 |                 if (tmp < 0) {
           |      |                    ~
           |      |                    |
           |      |                    (12) following 'false' branch (when 'tmp >= 0')...
           |......
           |  183 |                 memcpy(&(*result)[j], tmpres, tmp);
           |      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           |      |                 |
           |      |                 (13) ...to here
           |      |                 (14) use of uninitialized value '&tmpres' here
           |
../.././../src/v3.7.8-02e889560f.clean/lib/auth/srp_sb64.c: In function '_gnutls_sbase64_decode':
../.././../src/v3.7.8-02e889560f.clean/lib/auth/srp_sb64.c:281:17: warning: use of uninitialized value '&tmpres' [CWE-457] [-Wanalyzer-use-of-uninitialized-value]
  281 |                 memcpy(&(*result)[j], tmpres, tmp);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  'gnutls_srp_base64_decode2': events 1-2
    |
    |  430 | gnutls_srp_base64_decode2(const gnutls_datum_t * b64_data,
    |      | ^~~~~~~~~~~~~~~~~~~~~~~~~
    |      | |
    |      | (1) entry to 'gnutls_srp_base64_decode2'
    |......
    |  437 |             _gnutls_sbase64_decode((char *) b64_data->data, b64_data->size,
    |      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    |      |             |
    |      |             (2) calling '_gnutls_sbase64_decode' from 'gnutls_srp_base64_decode2'
    |  438 |                                    &ret);
    |      |                                    ~~~~~
    |
    +--> '_gnutls_sbase64_decode': events 3-10
           |
           |  237 | _gnutls_sbase64_decode(char *data, size_t idata_size, uint8_t ** result)
           |      | ^~~~~~~~~~~~~~~~~~~~~~
           |      | |
           |      | (3) entry to '_gnutls_sbase64_decode'
           |......
           |  243 |         uint8_t tmpres[3];
           |      |                 ~~~~~~
           |      |                 |
           |      |                 (4) region created on stack here
           |      |                 (5) capacity: 3 bytes
           |......
           |  254 |         if ((*result) == NULL)
           |      |            ~
           |      |            |
           |      |            (6) following 'false' branch...
           |......
           |  258 |         tmp = 0;
           |      |         ~~~~~~~
           |      |             |
           |      |             (7) ...to here
           |......
           |  275 |         for (i = left, j = tmp; i < idata_size; i += 4) {
           |      |                                 ~~~~~~~~~~~~~~
           |      |                                   |
           |      |                                   (8) following 'true' branch...
           |  276 |                 tmp = decode(tmpres, (uint8_t *) & data[i]);
           |      |                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           |      |                       |
           |      |                       (9) ...to here
           |      |                       (10) calling 'decode' from '_gnutls_sbase64_decode'
           |
           +--> 'decode': events 11-19
                  |
                  |  194 | inline static int decode(uint8_t * result, const uint8_t * data)
                  |      |                   ^~~~~~
                  |      |                   |
                  |      |                   (11) entry to 'decode'
                  |......
                  |  203 |         if (a1 != 0xff)
                  |      |            ~       
                  |      |            |
                  |      |            (12) following 'true' branch (when 'a1 != 255')...
                  |  204 |                 result[2] = a1 & 0xff;
                  |      |                       ~
                  |      |                       |
                  |      |                       (13) ...to here
                  |......
                  |  214 |         if (a2 != 0xff)
                  |      |            ~       
                  |      |            |
                  |      |            (14) following 'false' branch (when 'a2 == 255')...
                  |  215 |                 result[1] |= ((a2 & 0x0f) << 4);
                  |  216 |         else if (a1 == 0xff || result[1] == 0)
                  |      |                 ~  
                  |      |                 |
                  |      |                 (15) ...to here
                  |......
                  |  221 |         if (a1 != 0xff)
                  |      |            ~       
                  |      |            |
                  |      |            (16) following 'false' branch (when 'a1 == 255')...
                  |  222 |                 result[0] = (((a1 & 0x30) >> 4) & 0xff);
                  |  223 |         if (a2 != 0xff)
                  |      |            ~       
                  |      |            |
                  |      |            (17) ...to here
                  |      |            (18) following 'false' branch (when 'a2 == 255')...
                  |  224 |                 result[0] |= ((a2 << 2) & 0xff);
                  |  225 |         else if (a1 == 0xff || result[0] == 0)
                  |      |                 ~  
                  |      |                 |
                  |      |                 (19) ...to here
                  |
           <------+
           |
         '_gnutls_sbase64_decode': events 20-23
           |
           |  276 |                 tmp = decode(tmpres, (uint8_t *) & data[i]);
           |      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           |      |                       |
           |      |                       (20) returning to '_gnutls_sbase64_decode' from 'decode'
           |  277 |                 if (tmp < 0) {
           |      |                    ~   
           |      |                    |
           |      |                    (21) following 'false' branch (when 'tmp >= 0')...
           |......
           |  281 |                 memcpy(&(*result)[j], tmpres, tmp);
           |      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           |      |                 |
           |      |                 (22) ...to here
           |      |                 (23) use of uninitialized value '&tmpres' here
           |
../.././../src/v3.7.8-02e889560f.clean/lib/auth/srp_sb64.c:281:17: warning: use of uninitialized value '&tmpres' [CWE-457] [-Wanalyzer-use-of-uninitialized-value]
  281 |                 memcpy(&(*result)[j], tmpres, tmp);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  'gnutls_srp_base64_decode': events 1-2
    |
    |  388 | gnutls_srp_base64_decode(const gnutls_datum_t * b64_data, char *result,
    |      | ^~~~~~~~~~~~~~~~~~~~~~~~
    |      | |
    |      | (1) entry to 'gnutls_srp_base64_decode'
    |......
    |  395 |             _gnutls_sbase64_decode((char *) b64_data->data, b64_data->size,
    |      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    |      |             |
    |      |             (2) calling '_gnutls_sbase64_decode' from 'gnutls_srp_base64_decode'
    |  396 |                                    &res);
    |      |                                    ~~~~~
    |
    +--> '_gnutls_sbase64_decode': events 3-10
           |
           |  237 | _gnutls_sbase64_decode(char *data, size_t idata_size, uint8_t ** result)
           |      | ^~~~~~~~~~~~~~~~~~~~~~
           |      | |
           |      | (3) entry to '_gnutls_sbase64_decode'
           |......
           |  243 |         uint8_t tmpres[3];
           |      |                 ~~~~~~
           |      |                 |
           |      |                 (4) region created on stack here
           |      |                 (5) capacity: 3 bytes
           |......
           |  254 |         if ((*result) == NULL)
           |      |            ~
           |      |            |
           |      |            (6) following 'false' branch...
           |......
           |  258 |         tmp = 0;
           |      |         ~~~~~~~
           |      |             |
           |      |             (7) ...to here
           |......
           |  275 |         for (i = left, j = tmp; i < idata_size; i += 4) {
           |      |                                 ~~~~~~~~~~~~~~
           |      |                                   |
           |      |                                   (8) following 'true' branch...
           |  276 |                 tmp = decode(tmpres, (uint8_t *) & data[i]);
           |      |                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           |      |                       |
           |      |                       (9) ...to here
           |      |                       (10) calling 'decode' from '_gnutls_sbase64_decode'
           |
           +--> 'decode': events 11-19
                  |
                  |  194 | inline static int decode(uint8_t * result, const uint8_t * data)
                  |      |                   ^~~~~~
                  |      |                   |
                  |      |                   (11) entry to 'decode'
                  |......
                  |  203 |         if (a1 != 0xff)
                  |      |            ~       
                  |      |            |
                  |      |            (12) following 'true' branch (when 'a1 != 255')...
                  |  204 |                 result[2] = a1 & 0xff;
                  |      |                       ~
                  |      |                       |
                  |      |                       (13) ...to here
                  |......
                  |  214 |         if (a2 != 0xff)
                  |      |            ~       
                  |      |            |
                  |      |            (14) following 'false' branch (when 'a2 == 255')...
                  |  215 |                 result[1] |= ((a2 & 0x0f) << 4);
                  |  216 |         else if (a1 == 0xff || result[1] == 0)
                  |      |                 ~  
                  |      |                 |
                  |      |                 (15) ...to here
                  |......
                  |  221 |         if (a1 != 0xff)
                  |      |            ~       
                  |      |            |
                  |      |            (16) following 'false' branch (when 'a1 == 255')...
                  |  222 |                 result[0] = (((a1 & 0x30) >> 4) & 0xff);
                  |  223 |         if (a2 != 0xff)
                  |      |            ~       
                  |      |            |
                  |      |            (17) ...to here
                  |      |            (18) following 'false' branch (when 'a2 == 255')...
                  |  224 |                 result[0] |= ((a2 << 2) & 0xff);
                  |  225 |         else if (a1 == 0xff || result[0] == 0)
                  |      |                 ~  
                  |      |                 |
                  |      |                 (19) ...to here
                  |
           <------+
           |
         '_gnutls_sbase64_decode': events 20-23
           |
           |  276 |                 tmp = decode(tmpres, (uint8_t *) & data[i]);
           |      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           |      |                       |
           |      |                       (20) returning to '_gnutls_sbase64_decode' from 'decode'
           |  277 |                 if (tmp < 0) {
           |      |                    ~   
           |      |                    |
           |      |                    (21) following 'false' branch (when 'tmp >= 0')...
           |......
           |  281 |                 memcpy(&(*result)[j], tmpres, tmp);
           |      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           |      |                 |
           |      |                 (22) ...to here
           |      |                 (23) use of uninitialized value '&tmpres' here
           |
../../.././../src/v3.7.8-02e889560f.clean/lib/accelerated/x86/x86-common.c:1227:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]
 1227 | }
      | ^
x86_64-w64-mingw32-ar: coff/aesni-x86_64.o: No such file or directory
make[5]: *** [libx86.la] Error 1
make[4]: *** [all-recursive] Error 1
make[3]: *** [all-recursive] Error 1
make[2]: *** [all] Error 2
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```
</details>

**Comment by minnyres** [comment]
I also met the same problem with Debian 12 and llvm-mingw 16.0.6.

Error log for arm64-mingw-static triplet

> aarch64-w64-mingw32-llvm-ar: error: elf/sha1-armv8.o: No such file or directory
> make[5]: *** [Makefile:2202: libaarch64.la] Error 1
> make[4]: *** [Makefile:2251: all-recursive] Error 1
> make[3]: *** [Makefile:2903: all-recursive] Error 1
> make[2]: *** [Makefile:2507: all] Error 2
> make[1]: *** [Makefile:2262: all-recursive] Error 1

x64-mingw-static triplet

> x86_64-w64-mingw32-llvm-ar: error: coff/aesni-x86_64.o: No such file or directory
> make[5]: *** [Makefile:2365: libx86.la] Error 1
> make[4]: *** [Makefile:2251: all-recursive] Error 1
> make[3]: *** [Makefile:2903: all-recursive] Error 1
> make[2]: *** [Makefile:2507: all] Error 2
> make[1]: *** [Makefile:2262: all-recursive] Error 1
> make: *** [Makefile:2187: all] Error 2

**Comment by minnyres** [comment]
@Adela0814 this patch works for me

```patch
diff --git a/ports/libgnutls/portfile.cmake b/ports/libgnutls/portfile.cmake
index 784a81303..a05a3578a 100644
--- a/ports/libgnutls/portfile.cmake
+++ b/ports/libgnutls/portfile.cmake
@@ -33,6 +33,8 @@ if(VCPKG_TARGET_IS_OSX)
 endif()

 set(ENV{GTKDOCIZE} true) # true, the program
+set(ENV{CFLAGS} " -D_POSIX_C_SOURCE -Wno-int-conversion")
+set(ENV{CXXFLAGS} " -D_POSIX_C_SOURCE -Wno-int-conversion")
 vcpkg_configure_make(
     SOURCE_PATH "${SOURCE_PATH}"
     AUTOCONFIG
@@ -50,6 +52,7 @@ vcpkg_configure_make(
         --with-tpm=no
         --with-tpm2=no
         --with-zstd=no
+        --disable-hardware-acceleration
         ${options}
         YACC=false # false, the program - not used here
     OPTIONS_DEBUG
```

**Comment by dg0yt** [comment]
So the build fails due to lack of `coff/aesni-x86_64.o` (etc.)
And this is the command which would have created the file
~~~
libtool: compile:  : -g -c ../../.././../src/v3.7.8-02e889560f.clean/lib/accelerated/x86/coff/aesni-x86_64.s -o coff/aesni-x86_64.o
~~~
Note the command: `:` i.e. the no-op.
This should be the C compiler used for assembler, `CCAS`.

This issue was probably fixed by #34275, version 3.8.3#1.

**Comment by minnyres** [comment]
> This issue was probably fixed by #34275, version 3.8.3#1.

It works for triplet x64-mingw-static, while arm64-mingw-static still got an error

```
.././../src/v3.8.1-d0b7f6d114.clean/gl/canonicalize-lgpl.c:180:41: warning: result of comparison of constant 9223372036854775807 with expression of type 'long' is always true [-Wtautological-constant-out-of-range-compare]
  180 |   return path_max < 0 ? 1024 : path_max <= IDX_MAX ? path_max : IDX_MAX;
      |                                ~~~~~~~~ ^  ~~~~~~~
1 warning generated.
../.././../src/v3.8.1-d0b7f6d114.clean/lib/x509/output.c:465:40: warning: format specifies type 'unsigned long' but the argument has type 'time_t' (aka 'long long') [-Wformat]
  465 |                 addf(str, "error: gmtime_r (%lu)\n", timestamp);
      |                                             ~~~      ^~~~~~~~~
      |                                             %lld
../.././../src/v3.8.1-d0b7f6d114.clean/lib/x509/output.c:470:40: warning: format specifies type 'unsigned long' but the argument has type 'time_t' (aka 'long long') [-Wformat]
  470 |                 addf(str, "error: strftime (%lu)\n", timestamp);
      |                                             ~~~      ^~~~~~~~~
      |                                             %lld
2 warnings generated.
../.././../src/v3.8.1-d0b7f6d114.clean/lib/x509/verify-high2.c:480:43: warning: passing 'unsigned char *' to parameter of type 'const char *' converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Wpointer-sign]
  480 |                         ret = _gnutls_pathbuf_append(&pathbuf, utf8.data);
      |                                                                ^~~~~~~~~
../.././../src/v3.8.1-d0b7f6d114.clean/lib/x509/../pathbuf.h:42:19: note: passing argument to parameter 'component' here
   42 |                            const char *component);
      |                                        ^
1 warning generated.
../.././../src/v3.8.1-d0b7f6d114.clean/lib/auth/cert.c:390:55: warning: parameter 'data_size' set but not used [-Wunused-but-set-parameter]
  390 |                        int issuers_len, const uint8_t *data, size_t data_size)
      |                                                                     ^
1 warning generated.
lib/accelerated/aarch64/elf/sha1-armv8.s.tmp.S:6:1: error: unknown directive
.hidden _gnutls_arm_cpuid_s
^
lib/accelerated/aarch64/elf/sha1-armv8.s.tmp.S:8:7: error: expected absolute expression
.type sha1_block_data_order,%function
      ^
lib/accelerated/aarch64/elf/sha1-armv8.s.tmp.S:1076:1: error: unknown directive
.size sha1_block_data_order,.-sha1_block_data_order
^
lib/accelerated/aarch64/elf/sha1-armv8.s.tmp.S:1077:7: error: expected absolute expression
.type sha1_block_armv8,%function
      ^
lib/accelerated/aarch64/elf/sha1-armv8.s.tmp.S:1206:1: error: unknown directive
.size sha1_block_armv8,.-sha1_block_armv8
^
lib/accelerated/aarch64/elf/sha1-armv8.s.tmp.S:1222:19: error: unexpected token in directive
.section .note.GNU-stack,"",%progbits
                  ^
lib/accelerated/aarch64/elf/sha512-armv8.s.tmp.S:62:1: error: unknown directive
.hidden _gnutls_arm_cpuid_s
^
lib/accelerated/aarch64/elf/sha512-armv8.s.tmp.S:64:7: error: expected absolute expression
.type sha512_block_data_order,%function
      ^
make[5]: *** [Makefile:2327: elf/sha1-armv8.lo] Error 1
make[5]: *** Waiting for unfinished jobs....
lib/accelerated/aarch64/elf/sha512-armv8.s.tmp.S:1041:1: error: unknown directive
.size sha512_block_data_order,.-sha512_block_data_order
^
lib/accelerated/aarch64/elf/sha512-armv8.s.tmp.S:1044:7: error: expected absolute expression
.type .LK512,%object
      ^
lib/accelerated/aarch64/elf/sha512-armv8.s.tmp.S:1087:1: error: unknown directive
.size .LK512,.-.LK512
^
lib/accelerated/aarch64/elf/sha512-armv8.s.tmp.S:1101:7: error: expected absolute expression
.type sha512_block_armv8,%function
      ^
lib/accelerated/aarch64/elf/sha512-armv8.s.tmp.S:1617:1: error: unknown directive
.size sha512_block_armv8,.-sha512_block_armv8
^
lib/accelerated/aarch64/elf/sha512-armv8.s.tmp.S:1618:19: error: unexpected token in directive
.section .note.GNU-stack,"",%progbits
                  ^
make[5]: *** [Makefile:2327: elf/sha512-armv8.lo] Error 1
make[4]: *** [Makefile:2320: all-recursive] Error 1
make[3]: *** [Makefile:2975: all-recursive] Error 1
make[2]: *** [Makefile:2578: all] Error 2
make[1]: *** [Makefile:2325: all-recursive] Error 1
make: *** [Makefile:2250: all] Error 2

```

**Comment by dg0yt** [comment]
arm64-mingw is rarely tested, and probably unsupported by upstream. In that case, `-disable-hardware-acceleration` is fine - but it shall be set only for this known bad configuration, not for all targets.
